### PR TITLE
mkchromecast: init at 5872a246

### PR DESCRIPTION
**Pull request by bobvanderlinden**
###### Motivation for this change

This adds the mkchromecast package. i have tested this using:

```
nix-build -A pkgs.mkchromecast .
result/bin/mkchromecast --port 8080 --video --screencast --resolution 720p
```

And it would show my screen on my TV after a few moments.

The upgrade of pychromecast was not needed strictly speaking, but it seemed appropriate to enable all functionality of mkchromecast.

There wasn't a recent release, but that's something that hopefully will come (https://github.com/muammar/mkchromecast/issues/244).

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/misc/mkchromecast/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    rev    = version;
+    sha256 = "05ldgx583s4b3qqn2r3sj7wjmfdqndkm59g2bwdkpz7pbcahkfmr";
+  };
+
+  propagatedBuildInputs = [
+    PyChromecast psutil mutagen flask pyqt5 netifaces requests soco
+    pulseaudio vorbis-tools sox lame flac faac ffmpeg
+  ];
+
+  doCheck = false;
```
Comment (FRidh) [inline pkgs/applications/misc/mkchromecast/default.nix]: Why are the tests disabled? Include a comment in the expression explaining why the tests are disabled.

File: pkgs/applications/misc/mkchromecast/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
+{ lib, fetchFromGitHub, buildPythonApplication, pulseaudio
```
Comment (FRidh) [inline pkgs/applications/misc/mkchromecast/default.nix]: use `python3Packages` as argument instead

**Review by FRidh — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/mkchromecast/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+, PyChromecast, psutil, mutagen, flask, pyqt5, netifaces, requests, soco
+, vorbis-tools, sox, lame, flac, faac, ffmpeg }:
+buildPythonApplication rec {
+  pname = "mkchromecast";
+  version = "5872a246f0610b74fc2b197eb02dc91b96fb68cc";
```
Comment (etu) [inline pkgs/applications/misc/mkchromecast/default.nix]: Should be something human readable, like "unstable-2011-12-31" (change the date to the commit date of course)

**Comment by bobvanderlinden** [comment]
Sorry for the duplicate. This was apparently already submitted and the PR is in a further stage than this one: https://github.com/NixOS/nixpkgs/pull/57033

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -21930,6 +21930,10 @@ in
 
   m4rie = callPackage ../development/libraries/science/math/m4rie { };
 
+  mkchromecast = callPackage ../applications/misc/mkchromecast {
```
Comment (adisbladis) [inline pkgs/top-level/all-packages.nix]: I'd change this to:
``` nix
mkchromecast = python3Packages.callPackage ../applications/misc/mkchromecast { };
```
So you don't have to pass each `python3Packages` attribute explicitly.

**Pull request closed without merging**